src/database/transaction.rs

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use std::{fmt, marker::PhantomData, sync::Arc};
+
+use super::query::QueryManager;
+use crate::{
+    common::{Result, TransactionType},
+    connection::TransactionStream,
+    Options,
+};
+
+pub struct Transaction<'a> {
+    type_: TransactionType,
+    options: Options,
+
+    query: QueryManager,
+    transaction_stream: Arc<TransactionStream>,
+
+    _lifetime_guard: PhantomData<&'a ()>,
```

Comment:
`PhantomData` allows us to enforce the lifetime dependency even when there isn't an actual rust-native dependency in sight. The compiler essentially "pretends" that a `Transaction<'a>` owns a reference with lifetime `'a` and proceeds accordingly.